genet/core.py

```diff
@@ -432,6 +436,100 @@ def links_on_spatial_condition(self, region_input, how='intersect'):
             # is assumed to be hex
             return self._find_link_ids_on_s2_geometry(gdf, how, region_input)
 
+    def subnetwork(self, links: Union[list, set], services: Union[list, set] = None):
+        """
+        Subset a Network object using a collection of link IDs and (optionally) service IDs
+        :param links: Link IDs to be retained in the new Network
+        :param services: optional, collection of service IDs in the Schedule for subsetting.
+        :return: A new Network object that is a subset of the original
+        """
+        logging.info('Subsetting a Network will likely result in a disconnected network graph. A cleaner will be ran '
+                     'that will remove links to make the resulting Network strongly connected for modes: '
+                     'car, walk, bike.')
```

Comment:
Should this be parametrised for future compatibility? Perhaps not, as car, walk, bike is fairly standard and I can't think of any non PT modes we may introduce in the future. I suppose networks for scooters etc is technically possible.